etna/auto/auto.py

```diff
@@ -361,26 +478,108 @@ def _objective(trial: Trial) -> float:
 
         return _objective
 
-    def _init_optuna(self):
+    def _init_pool_optuna(self, suppress_logging: bool = False) -> Optuna:
         """Initialize optuna."""
-        if isinstance(self.pool, Pool):
-            pool: List[Pipeline] = self.pool.value.generate(horizon=self.horizon)
-        else:
-            pool = self.pool
+        pool = [pipeline.to_dict() for pipeline in self._pool]
+        logging_verbosity = optuna.logging.get_verbosity()
+        try:
```

Comment:
What is the purpose of using try finally?

Reply:
If smth goes wrong in creating an `Optuna` object we will return verbosity of optuna as it was before. Otherwise it will be changed as a side-effect.

Reply:
Let's imagine the situation: creation of `Optuna` object fails, because storage is incorrect. In this case we set `optuna.logging.set_verbosity` to a new value and don't change it back. The logs changed its verbosity forever unless user changes it back like he wants.

Reply:
`creation of Optuna object fails` - is it a situation we can pass without any handling? We'll return `pool_optuna` which is not defined

Reply:
In a given code if `Optuna` creation fails then we make code in `finally` block and fail after that, we don't return anything.

Reply:
I see:

```
return pool_optuna
```

Reply:
But we return  `Optuna` as for function signature

Reply:
I've forgotten finally behaviour. Sorry